src/DynamicMvvm.Abstractions/ViewModel/IViewModel.Extensions.Children.cs

```diff
@@ -127,6 +127,29 @@ public static void DetachChild(this IViewModel viewModel, IViewModel childViewMo
 			viewModel.RemoveDisposable(name ?? childViewModel.Name);
 		}
 
+		/// <summary>
+		/// Attaches a child <see cref="IViewModel"/> to a parent <see cref="IViewModel"/>.
+		/// If the child has already been attached, the newer instance will be attached instead of the previous instance.
+		/// By being attached, the child will be disposed when the parent is disposed.
+		/// Both also share the same <see cref="IViewModelView"/>.
+		/// </summary>
+		/// <remarks>The previous instance will be disposed when detached.</remarks>
+		/// <param name="viewModel">The parent <see cref="IViewModel"/>.</param>
+		/// <param name="childViewModel">The child <see cref="IViewModel"/> to detach.</param>
+		/// <param name="name">The child ViewModel's name. This defaults to <paramref name="childViewModel"/>.Name when not provided.</param>
+		public static TChildViewModel AttachOrReplaceChild<TChildViewModel>(this IViewModel viewModel, TChildViewModel childViewModel, string name = null)
+			where TChildViewModel : IViewModel
+		{
+			name = name ?? childViewModel.Name;
+
+			if (viewModel.TryGetDisposable(name, out var instance))
```

Comment:
Just curious in what kind of situation the TryGetDisposable can fail to return an instance?

Reply:
I think in this case : https://github.com/nventive/Chinook.DynamicMvvm/blob/09b12490a11d7e579c65b1b965ec211183163231/src/DynamicMvvm/ViewModel/ViewModelBase.Disposable.cs#L48